homeassistant/helpers/entity_component.py

```diff
@@ -172,24 +172,13 @@ def setup(self, config):
     async def async_extract_from_service(self, service, expand_group=True):
         """Extract all known and available entities from a service call.
 
-        Will return all entities if no entities specified in call.
         Will return an empty list if entities specified but unknown.
 
         This method must be run in the event loop.
         """
         data_ent_id = service.data.get(ATTR_ENTITY_ID)
 
-        if data_ent_id in (None, ENTITY_MATCH_ALL):
-            if data_ent_id is None:
-                self.logger.warning(
-                    "Not passing an entity ID to a service to target all "
-                    "entities is deprecated. Update your call to %s.%s to be "
-                    "instead: entity_id: %s",
-                    service.domain,
-                    service.service,
-                    ENTITY_MATCH_ALL,
-                )
-
+        if data_ent_id in (ENTITY_MATCH_ALL):
```

Comment:
this is a bug. you now check if `data_ent_id` is a valid subscring of `ENTITY_MATCH_ALL`

```suggestion
        if data_ent_id == ENTITY_MATCH_ALL:
```